### PR TITLE
SIL ownership staging

### PR DESCRIPTION
**Pull request by gottesmm**
This PR contains the staging necessary for hiding semantic arc changes from affecting other developers in tree.

It is the first part of Semantic ARC that needs to land so we can create bots.

rdar://28685236

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci Please smoke test

File: include/swift/AST/SILOptions.h [hunk 1]
```diff
@@ -113,6 +113,9 @@ class SILOptions {
 
   /// The name of the SIL outputfile if compiled with SIL debugging (-gsil).
   std::string SILOutputFileNameForDebugging;
+
+  /// If set to true, compile with the SIL Ownership Model enabled.
+  bool EnableSILOwnershipModel = false;
```
Comment (atrick) [inline include/swift/AST/SILOptions.h]: "EnableSILOwnership" is sufficient.

**Review by atrick — COMMENTED** [review]
👍

File: include/swift/AST/SILOptions.h (hunk 1 above)
Comment (gottesmm) [inline include/swift/AST/SILOptions.h]: Done

**Comment by gottesmm** [comment]
@swift-ci Please smoke test

**Comment by gottesmm** [comment]
@swift-ci Please smoke test linux platform

**Pull request merged**